probe/endpoint/ebpf.go

```diff
@@ -0,0 +1,226 @@
+package endpoint
+
+import (
+	"bufio"
+	"net"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"sync"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+// TCPV4TracerLocation is the location of the Python script
+// that delivers the eBPF messages coming from the kernel.
+var TCPV4TracerLocation = "/home/weave/tcpv4tracer.py"
+
+// A ebpfConnection represents a network connection
+type ebpfConnection struct {
+	tuple            fourTuple
+	networkNamespace string
+	incoming         bool
+	pid              int
+}
+
+type eventTracker interface {
+	handleFlow(eventType string, tuple fourTuple, pid int, networkNamespace string)
+	hasDied() bool
+	run()
+	walkFlows(f func(ebpfConnection))
+	initialize()
+	isInitialized() bool
+}
+
+// nilTracker is a tracker that does nothing, and it implements the eventTracker interface.
+// It is returned when the ebpfEnabled flag is false.
+type nilTracker struct{}
+
+func (n nilTracker) handleFlow(_ string, _ fourTuple, _ int, _ string) {}
+func (n nilTracker) hasDied() bool                                     { return true }
+func (n nilTracker) run()                                              {}
+func (n nilTracker) walkFlows(f func(ebpfConnection))                  {}
+func (n nilTracker) initialize()                                       {}
+func (n nilTracker) isInitialized() bool                               { return false }
+
+// EbpfTracker contains the list of eBPF events, and the eBPF script's command
+type EbpfTracker struct {
+	sync.Mutex
+	cmd *exec.Cmd
+
+	initialized bool
+	dead        bool
+
+	activeFlows   map[string]ebpfConnection
+	bufferedFlows []ebpfConnection
```

Comment:
ok

Can you add a comment saying that they are recently closed connections and they are not staying forever here but only until the next walkConnections